siliconcompiler/core.py

```diff
@@ -3889,24 +3901,41 @@ def run(self):
         # Merge cfg back from last executed runsteps.
         # Note: any information generated in steps that do not merge into the
         # last step will not be picked up in this chip object.
+
+        #1. Use the last manifest
         laststep = steplist[-1]
         last_step_failed = True
-        for index in indexlist[laststep]:
-            lastdir = self._getworkdir(step=laststep, index=index)
-
-            # This no-op listdir operation is important for ensuring we have a
-            # consistent view of the filesystem when dealing with NFS. Without
-            # this, this thread is often unable to find the final manifest of
-            # runs performed on job schedulers, even if they completed
-            # successfully. Inspired by: https://stackoverflow.com/a/70029046.
-            os.listdir(os.path.dirname(lastdir))
-
-            lastcfg = f"{lastdir}/outputs/{self.get('design')}.pkg.json"
-            if os.path.isfile(lastcfg):
-                last_step_failed = False
-                local_dir = self.get('dir')
-                self.read_manifest(lastcfg, clobber=True, clear=True)
-                self.set('dir', local_dir)
+        lastdir = self._getworkdir(step=laststep, index=index)
```

Comment:
Be careful here, I think you might want to hard code `index` to 0. I think the most recent reference to 'index' comes in the for loop that sets up multiprocessing, so the value could end up being something weird.